driver-core/src/test/unit/com/mongodb/operation/BulkWriteBatchSpecification.groovy

```diff
@@ -41,8 +44,13 @@ import static com.mongodb.connection.ServerConnectionState.CONNECTED
 
 class BulkWriteBatchSpecification extends Specification {
     def namespace = new MongoNamespace('db.coll')
-    def serverDescription = ServerDescription.builder().address(new ServerAddress()).state(CONNECTED).build()
-    def connectionDescription = new ConnectionDescription(new ServerId(new ClusterId(), serverDescription.getAddress()))
+    def serverDescription = ServerDescription.builder().address(new ServerAddress()).state(CONNECTED)
```

Comment:
Why are these changes needed?

Reply:
Ah, good question.  I should have commented on that. Without this change OperationHelper#canRetryWrite returns false and the test passes even if the code change is not made.  I almost missed that because I changed the code before writing the test, but just to be sure I reverted the code change and re-ran the test and was surprised to see that it still passed.  A good less for me to do TDD religiously even for small changes like this.